README.md

```diff
@@ -522,72 +525,169 @@ To add a `saml:AuthnContextDeclRef`, define `settings.authn_context_decl_ref`.
 In a SP-initiated flow, the SP can indicate to the IdP the subject that should be authenticated. This is done by defining the `settings.name_identifier_value_requested` before
 building the authrequest object.
 
+## Service Provider Metadata
+
+To form a trusted pair relationship with the IdP, the SP (you) need to provide metadata XML
+to the IdP for various good reasons. (Caching, certificate lookups, relaying party permissions, etc)
 
-## Signing
+The class `OneLogin::RubySaml::Metadata` takes care of this by reading the Settings and returning XML.  All you have to do is add a controller to return the data, then give this URL to the IdP administrator.
+
+The metadata will be polled by the IdP every few minutes, so updating your settings should propagate
+to the IdP settings.
 
-The Ruby Toolkit supports 2 different kinds of signature: Embeded and `GET` parameters
+```ruby
+class SamlController < ApplicationController
+  # ... the rest of your controller definitions ...
+  def metadata
+    settings = Account.get_saml_settings
+    meta = OneLogin::RubySaml::Metadata.new
+    render :xml => meta.generate(settings), :content_type => "application/samlmetadata+xml"
+  end
+end
+```
 
-In order to be able to sign, define the private key and the public cert of the service provider:
+You can add `ValidUntil` and `CacheDuration` to the SP Metadata XML using instead:
 
 ```ruby
-  settings.certificate = "CERTIFICATE TEXT WITH HEAD AND FOOT"
-  settings.private_key = "PRIVATE KEY TEXT WITH HEAD AND FOOT"
+  # Valid until => 2 days from now
+  # Cache duration = 604800s = 1 week
+  valid_until = Time.now + 172800
+  cache_duration = 604800
+  meta.generate(settings, false, valid_until, cache_duration)
 ```
 
-The settings related to sign are stored in the `security` attribute of the settings:
+## Signing and Encryption
+
+Ruby SAML supports supports the signing and encryption functionality:
+
+1. Signing your SP Metadata XML
+2. Signing your SP SAML messages
+3. Encrypting IdP Assertion messages, and decrypting them upon receipt (EncryptedAssertion)
```

Comment:
Toolkit does not support  "Encrypting IdP Assertion messages", only decrypt them